spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/security/saml2/Saml2RelyingPartyAutoConfigurationTests.java

```diff
@@ -61,18 +63,20 @@ class Saml2RelyingPartyAutoConfigurationTests {
 	private final WebApplicationContextRunner contextRunner = new WebApplicationContextRunner().withConfiguration(
 			AutoConfigurations.of(Saml2RelyingPartyAutoConfiguration.class, SecurityAutoConfiguration.class));
 
-	@Test
-	void autoConfigurationShouldBeConditionalOnRelyingPartyRegistrationRepositoryClass() {
-		this.contextRunner.withPropertyValues(getPropertyValues()).withClassLoader(new FilteredClassLoader(
+	@ParameterizedTest
+	@ValueSource(booleans = { true, false })
+	void autoConfigurationShouldBeConditionalOnRelyingPartyRegistrationRepositoryClass(boolean useDeprecated) {
+		this.contextRunner.withPropertyValues(getPropertyValues(useDeprecated)).withClassLoader(new FilteredClassLoader(
 				"org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository"))
 				.run((context) -> assertThat(context).doesNotHaveBean(RelyingPartyRegistrationRepository.class));
 	}
 
-	@Test
-	void autoConfigurationShouldBeConditionalOnServletWebApplication() {
+	@ParameterizedTest
+	@ValueSource(booleans = { true, false })
```

Comment:
A bit odd. I'd rather have two separate tests (and the one using the deprecated form should be marked as deprecated so that we can clean things up once we remove the deprecated feature that it tests.